src/api2/api-caller.js

```diff
@@ -0,0 +1,180 @@
+import axios from 'axios'
+
+import { Network, TransactionBuilder } from '../base'
+import { Wallet } from '../wallet'
+
+import middlewares from './middlewares'
+
+const SUBMIT_TRANSACTION_TIMEOUT = 60 * 10000
+
+const methods = Object.freeze({
+  PATCH: 'PATCH',
+  POST: 'POST',
+  PUT: 'PUT',
+  GET: 'GET'
+})
+
+/**
+ * ApiCaller performs the request to the API, with the following
+ */
+export class ApiCaller {
+  /**
+   * @param {object} opts
+   * @param {AxiosInstance} opts.axios
+   * @param {string} opts.baseURL
+   *
+   * @param {Wallet} [opts.wallet] - the initialized {@link Wallet} instance for signing requests/transactions
+   * @param {string} [opts.passphrase] - the passphrase of current TokenD network (is used internally when signing transactions)
+   */
+  constructor (opts) {
+    this._axios = opts.axios
+    this._baseURL = opts.baseURL
+
+    this._wallet = null
+    this._customTimeout = null
+    this._clockDiff = 0
+
+    if (opts.wallet) {
+      this.useWallet(opts.wallet)
+    }
+
+    if (opts.passphrase) {
+      this.usePassphrase(opts.passphrase)
+    }
+  }
+
+  static getInstance (baseURL) {
+    return new ApiCaller({
+      baseURL,
+      axios: axios.create()
+    })
+  }
+
+  static async getInstanceWithPassphrase (baseURL) {
+    const caller = this.getInstance(baseURL)
+    const networkDetails = await caller.get('/')
+
+    caller.usePassphrase(networkDetails.data.networkPassphrase)
+
+    return caller
+  }
+
+  patch (endpoint, query) { return this._call(endpoint, query, methods.PATCH) }
+  post (endpoint, query) { return this._call(endpoint, query, methods.POST) }
+  put (endpoint, query) { return this._call(endpoint, query, methods.PUT) }
+  get (endpoint, query) { return this._call(endpoint, query, methods.GET) }
+
+  getWithSignature (endpoint, query) {
+    return this._callWithSignature(endpoint, query, methods.GET)
+  }
+
+  patchWithSignature (endpoint, query) {
+    return this._callWithSignature(endpoint, query, methods.PATCH)
+  }
+
+  postWithSignature (endpoint, query) {
+    return this._callWithSignature(endpoint, query, methods.POST)
+  }
+
+  putWithSignature (endpoint, query) {
+    return this._callWithSignature(endpoint, query, methods.PUT)
+  }
+
+  /**
+   * Craft the transaction, sign it and make the post request with it's
+   * enveloper to the backend
+   * @param operations
+   * @returns {Promise<*>}
+   */
+  postTx (...operations) {
```

Comment:
please better describe that the method accepts the set of operations and builds them to TX to avoid confusing with the `postTxEnvelope` below